### PR TITLE
Refactored the way of the activity to get the gps status.

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/53" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public static final String GPS_STATUS_EXTRA = "gpsStatusExtra";
+
+    public static void broadcastGpsStatus(Context context, boolean gpsStatus) {
+        Intent intent = new Intent(BROADCAST_GPS_STATUS_ACTION);
+        intent.putExtra(GPS_STATUS_EXTRA, gpsStatus);
+
+        LocalBroadcastManager.getInstance(context).sendBroadcast(intent);
+    }
+
+    private Context mApplicationContext;
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java:37]: change the location of the private memebers (should be after the public static and before the functions, unless we declared different thing in the conventions)

File: app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private ConsistentStatusEventRaiser<Boolean> mEventRaiser;
+
+    public GpsStatusBroadcastReceiver(Context applicationContext)  {
+        mApplicationContext = applicationContext;
+
+        // TODO: Config
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java:43]: todo config - so do config

File: app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     * An intent action name, that the BL class should register to
+     */
+    public static final String BROADCAST_NEW_GPS_STATUS_ACTION = "broadCastNewGpsStatus";
+
+    /**
+     * The extra string key
+     */
+    public static final String GPS_STATUS_EXTRA = "gpsStatusExtra";
+
+    public static void broadcastGpsStatus(Context context, boolean gpsStatus) {
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java:30]: the broadcast receiver usually do both? or its seperated to receiver and broadcast?

File: app/src/main/res/values/config.xml [hunk 4]
```diff
@@ -5,7 +5,8 @@
 
     <!--location fetcher configuration-->
     <integer name="location_min_update_frequency_ms">1000</integer>
-    <integer name="location_threshold_update_distance_m">3</integer>
+    <integer name="location_threshold_update_distance_m">10</integer>
```
Comment (gRaytan) [inline app/src/main/res/values/config.xml:8]: should is be alspo in the debug config xml?

File: app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java (hunk 2 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java:43]: It hasn't been pushed to develop yet

File: app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java (hunk 1 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/receivers/GpsStatusBroadcastReceiver.java:37]: We agreed on this order: static - public, private, public methods, private methods. non-static - public, private, constructor, public methods, private methods

**Pull request merged**